script/sdcc_map.py

```diff
@@ -0,0 +1,198 @@
+#!/usr/bin/python3
+
+usage = """python sdcc_map.py <filename.map>
+ * Parses an sdcc linker .map file and outputs symbol, section, module 
+   information in .csv format
+ * Requires python 3.8+
+"""
+
+import sys,os
+import re
+from types import SimpleNamespace
+
+class Symbol:
+    def __init__(self, name, address, module, section):
+        self.name = name
+        self.address = address
+        self.module = module
+        self.section = section
+        self.size = -1
+
+class Module:
+    def __init__(self, name):
+        self.name = name
+        self.symbols = {}
+        self.size = 0
+
+    def addSymbol(self, symbol):
+        self.symbols[symbol.name] = symbol
+
+    def process(self):
+        self.size = 0
+        for (n,s) in self.symbols.items():
+            self.size += s.size
+
+class Section:
+    def __init__(self, name, address, size, attributes):
+        self.name =name
+        self.address = address
+        self.size = size
+        self.attributes = attributes
+        self.symbols = []
+
+    def begin(self):
+        return self.address
+
+    def end(self):
+        return self.address+self.size
+
+    def process(self):
+        if len(self.symbols) <= 0:
+            return
+
```

Comment:
using the section end as basis for calculation, as done in the case of the single symbol is actually better overall:
it gives us correct values for the symbols at the end of the section.
propose we rewrite this function as 
```
    def process(self):
        if len(self.symbols) <= 0:
            return

        self.symbols.sort(key=lambda x: x.address, reverse=True)

        sectionEnd = self.end()
        for symbol in self.symbols:
            symbol.size = sectionEnd - symbol.address
            sectionEnd = sectionEnd - symbol.size
```